### PR TITLE
feat(ui): React 18

### PR DESCRIPTION
**Pull request by scttcper**
fixes https://github.com/getsentry/frontend-tsc/issues/22

### CODE REVIEW

File: static/app/bootstrap/exportGlobals.tsx [hunk 1]
```diff
@@ -16,7 +17,7 @@ const globals = {
   Sentry,
   moment,
   Router: ReactRouter,
-  ReactDOM: {findDOMNode, render},
```
Comment (scttcper) [inline static/app/bootstrap/exportGlobals.tsx:19]: i have no idea what this is

File: jest.config.ts [hunk 2]
```diff
@@ -233,7 +233,6 @@ const config: Config.InitialOptions = {
   setupFilesAfterEnv: [
     '<rootDir>/tests/js/setup.ts',
     '<rootDir>/tests/js/setupFramework.ts',
-    '@testing-library/jest-dom/extend-expect',
```
Comment (scttcper) [inline jest.config.ts:236]: the export has moved to `@testing-library/jest-dom` and is imported in setup.ts

File: static/app/bootstrap/renderDom.tsx [hunk 3]
```diff
@@ -13,5 +13,6 @@ export function renderDom(
     return;
   }
 
-  render(<Component {...props} />, rootEl);
+  const root = createRoot(rootEl);
```
Comment (JonasBa) [inline static/app/bootstrap/renderDom.tsx:16]: Does this mean we opt into the new concurrent renderer as well?

Comment (scttcper) [reply static/app/bootstrap/renderDom.tsx:16]: currently that's the plan

Comment (scttcper) [reply static/app/bootstrap/renderDom.tsx:16]: okay the new plan is to put it behind an option

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**